src/LiveComponent/tests/Integration/LiveComponentHydratorTest.php

```diff
@@ -324,6 +326,111 @@ public function onEntireEntityUpdated($oldValue)
             ;
         }];
 
+        yield 'DTO with Doctrine ArrayCollection with persisted entities: (de)hydration works correctly' => [function () {
+            $firstProduct = create(ProductFixtureEntity::class, ['id' => 1, 'name' => 'first'])->object();
+            \assert($firstProduct instanceof ProductFixtureEntity);
+
+            $secondProduct = create(ProductFixtureEntity::class, ['id' => 2, 'name' => 'second'])->object();
+            \assert($secondProduct instanceof ProductFixtureEntity);
+
+            $productList = new ArrayCollection([$firstProduct, $secondProduct]);
+            \assert($productList instanceof ArrayCollection);
+
+            $doctrineEntityDto = create(HoldsArrayCollectionAndEntity::class)->object();
+            $doctrineEntityDto->setProduct($firstProduct);
+            $doctrineEntityDto->setProductList($productList);
+            \assert($doctrineEntityDto instanceof HoldsArrayCollectionAndEntity);
+
+            return HydrationTest::create(new class() {
+                #[LiveProp()]
+                public HoldsArrayCollectionAndEntity $doctrineEntityDto;
+            })
+                ->mountWith(['doctrineEntityDto' => $doctrineEntityDto])
+                ->assertDehydratesTo([
+                    'doctrineEntityDto' => [
+                        'product' => $firstProduct->id,
+                        'productList' => [
+                            [
+                                'className' => $firstProduct::class,
```

Comment:
Thus far, we've avoided sending PHP class names to the frontend. For example, an `array($product1, $product2)` is already supported, as long as the `LiveProp` that this is attached to properly documents that this is `@var Product[]`. That logic, iirc, for hydration is this: https://github.com/symfony/ux/blob/2.x/src/LiveComponent/src/LiveComponentHydrator.php#L267-L272 - and here is the fixture class that we use in the tests - https://github.com/symfony/ux/blob/2.x/src/LiveComponent/tests/Integration/LiveComponentHydratorTest.php#L1643-L1648

Not exposing PHP class info to the frontend might be paranoid. But until we get a big need for it, I'd like to continue avoiding it.

But we should definitely support your situation. Does it already work if you add documentation above the property to say that this is a collection - e.g. something like:

```php
class ItemsDTO
{
    /** @var Item[] */
    private ArrayCollection $items;
}
```

If this DOES already work, then what we need is a better error: if we detect that a property can't be hydrated, but it is some sort of iterable of objects, then we can tell them extra documentation is needed. If this does *not* currently work, then we should make it work - but the change might be inside `LiveComponentHydrator`.

Reply:
@weaverryan thanks for your reply 

I just tested it and indeed with the annotation and this work like a charm
```php
class ItemsDTO
{
    /** @var Item[] */
    private array $items;
}
```

The ArrayCollection version does not work.
```php
class ItemsDTO
{
    /** @var Item[] */
    private ArrayCollection $items;
}
```

This generates the following error. "Error rendering "form_name" component: Can't get a way to read the property "elements" in class "Doctrine\Common\Collections\ArrayCollection"."

It would be nice to have a clearer error message in this situation.

This PR is not needed anymore :)

Reply:
Yay!

> It would be nice to have a clearer error message in this situation.

I agree! It feels like, a bit earlier, we should check to see if the property is an iterable. And, if so, throw a better exception. I'd love a PR if you can make one - you obviously understand the technical details quite well.